### DISCUSSION TITLE
Connect multiple Outputs of Circuits with tristate Pins

### DISCUSSION THREAD

**Discussion by leonuetzel** [flandreas/antares#307] [Q&A] [answered]
Hello,

how is it possible to connect the Outputs of several Circuits with Output Pins that can be Tristate?
I got a logic that only one of those Outputs will be enabled at once, but Antares wont let me connect those Outputs together to the Circuit Output, because it doesnt recognize that only one of those Outputs will be active at once. How did you create the "Tristate Buffer" Circuit in the Standard Lib to achieve this Functionality?

**Comment by flandreas** [comment]
@leonuetzel I think you've raised an interesting point here. Also related to what people were discussing in #155.

The output of the "Tristate Buffer" in the standard library is internally marked as "can be undefined" aka "tristate" aka "floating". So is e.g. the D output of the ROM component.

It's funny that I myself never came across the need to be able to mark an output of a custom sub circuit as "tristate". It seems like in every sub circuit I've ever designed those ports where bi-directional, and those ports you can always connect however you like.

And of course, as long as Antares tries to guide the user in avoiding to connect outputs to outputs where this wouldn't make sense, Antares can't detect in which situations it DOES make sense (at least not easily), unless the user explicitly declares it.

So, I think that's worth a new feature: Let the user mark circuit outputs as "tristate". You would then be able to connect an input pin (e.g. of a circuit output) with many of such "tristate" outputs.

Would this solve your current problem?

**Comment by leonuetzel** [comment]
Yeah, thats exactly what i need :)

**Comment by flandreas** [comment, marked as answer]
Turned into feature issue #309

**Reply by siddhc** [reply]
@flandreas @leonuetzel My eventual plan is to build an 8-bit computer on Antares. So, this topic (i.e. _Connect multiple Outputs of Circuits with tristate Pins_) is a very important area for me presently. I made three 8-bit registers connected to a bus. The bus is also connected to a 8-pin dip switch. This dip switch loads data and then the three registers load it and shuffle it between themselves. I enclose the project here. It is working. However it will be very helpful if you can have a look at it and suggest some improvement areas. 
[Bus & Register Test.zip](https://github.com/flandreas/antares/files/9844358/Bus.Register.Test.zip)

**Reply by flandreas** [reply]
@siddhc I'm not an expert, but your design looks good to me.

Your transceiver tri-state gates are currently outside the registers. That's fine, and that's the reason why you currently don't need the property "Tri-State Output" (as of issue #309) in the output component O of your 8 bit register. If you were to move the transceiver logic into the registers, then you would have to set "Tri-State Output" on O in order to be able to connect many registers to the same bus.

Besides that, I only mention a few little Antares usage things:
- I would avoid placing components over wires, as the LOAD wire leading to the tri-state gate. This confused me for a short moment.
- You've exported your circuits using Circuit->Export, which includes all dependent circuits as well. This is fine. However, a circuit bundle can only be imported into an existing project. You could have also exported your project using the projects dialog (File -> Projects -> Export), which creates a project/library bundle. This can be imported into a separate project by the receiver of your export.
- I've noticed that your LED and switch require an edged wire. Just to make sure that you are aware that you can rotate components, e.g. by applying Cmd-R on a selected component.
- You have a port named ~Q. That's okay. You could also use the notation !Q, which will be displayed by Antares using an overline over Q.

**Reply by siddhc** [reply]
@flandreas Thank for your comments and helpful tips. 
Loved the ~Q/!Q one, did wonder about it yesterday. :)
